e2e_test/batch/order/test_order_requirers_index.part

```diff
@@ -0,0 +1,323 @@
+# description: Test lookup joins with order
+
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
+
+statement ok
+create table t1 (a int, b int, c int);
+
+statement ok
+INSERT INTO t1 VALUES (10, 0, 35), (10, -2, 28), (-2, 1, 81), (6, -5, 29), (-2, -5, 51), (8, 2, 17), (1, 4, 54), (10, 5, 53), (-1, 3, 96), (10, -1, 70), (5, 2, 50), (-10, 3, 55), (9, -2, 32), (-8, 1, 91), (-2, -1, 75), (9, 4, 13), (10, -1, 82), (-3, -2, 14), (-2, 2, 24), (-5, -3, 58), (-1, 5, 35), (5, 2, 13), (-10, -5, 11), (-9, -4, 39), (2, 4, 31), (-3, 2, 77), (5, 5, 10), (-10, 2, 19), (-9, 2, 49), (2, -5, 16), (4, 1, 66), (-1, -4, 35), (-8, 5, 75), (9, -4, 100), (-10, 4, 51), (-2, 0, 36), (-10, 5, 80), (6, 2, 79), (-7, -4, 91), (-1, 1, 76), (4, 2, 100), (8, 2, 44), (-5, 2, 13), (5, -1, 83), (9, 2, 34), (1, 0, 45), (-5, -5, 50), (6, 0, 71), (3, -4, 58), (-4, -4, 46);
+
+statement ok
+create index idx1 on t1(a, b, c);
+
+# Index on all required columns
+query I rowsort
+select * from t1 order by a;
+----
+-10 3 55
+-10 2 19
+-10 5 80
+-10 4 51
+-10 -5 11
+-9 2 49
+-9 -4 39
+-8 1 91
+-8 5 75
+-7 -4 91
+-5 -3 58
+-5 2 13
+-5 -5 50
+-4 -4 46
+-3 -2 14
+-3 2 77
+-2 -5 51
+-2 1 81
+-2 0 36
+-2 -1 75
+-2 2 24
+-1 1 76
+-1 -4 35
+-1 5 35
+-1 3 96
+1 0 45
+1 4 54
+2 -5 16
+2 4 31
+3 -4 58
+4 1 66
+4 2 100
+5 -1 83
+5 2 50
+5 5 10
+5 2 13
+6 0 71
+6 -5 29
+6 2 79
+8 2 17
+8 2 44
+9 -2 32
+9 -4 100
+9 2 34
+9 4 13
+10 -1 82
+10 5 53
+10 0 35
+10 -1 70
+10 -2 28
+
+
+statement ok
+drop index idx1;
+
+# Index includes all required columns
+
+statement ok
+create index idx1 on t1(b) include(a, c);
+
+query I rowsort
+select * from t1 order by b;
```

Comment:
There seems to be a problem in this case, as the output is not sorted by b.
Is it possible that this is an error in the implementation, or did I make wrong assumptions here?
@st1page